cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -817,6 +824,140 @@ def ceph_storage(ceph_admin):
     set_state('ceph-storage.configured')
 
 
+@when('config.changed.default-storage')
+def render_ceph():
+    remove_state('ceph-storage.configured')
+
+
+@when('ceph-storage.available')
+@when('cdk-addons.configured')
+@when_not('ceph-storage.configured')
+def ceph_csi_storage(ceph_admin):
+    '''Ceph on kubernetes will require a few things put on the cluster.
+    This includes a provisioner and storage class. This method will
+    put those things on the cluster.'''
+
+    # Pre-CSI versions still use the old apt version and manual volume
+    # creation, so skip the CSI stuff
+    if get_version('kube-apiserver') < (1, 10):
+        return
+
+    if ceph_admin.key():
+        encoded_key = base64.b64encode(ceph_admin.key().encode('utf-8'))
+    else:
+        # We didn't have a key, and cannot proceed. Do not set state and
+        # allow this method to re-execute
+        return
+
+    ceph_context = {
+        'mon_hosts': ceph_admin.mon_hosts(),
+        'fsid': ceph_admin.fsid(),
+        'auth_supported': ceph_admin.auth(),
+        'use_syslog': "true",
+        'ceph_public_network': '',
+        'ceph_cluster_network': '',
+        'loglevel': 1,
+        'hostname': socket.gethostname(),
+        'admin_key': encoded_key.decode('ascii'),
+        'kubernetes_key': encoded_key.decode('ascii'),
+    }
+
+    # Install the ceph common utilities.
+    apt_install(['ceph-common'], fatal=True)
```

Comment:
Is anything using the `ceph-common` package? It looks like it might be a carry-over from the old ceph stuff that we don't need anymore.